### ISSUE TITLE
Add function-ability testing for Windows (Travis CI or GitHub Actions)

### ISSUE THREAD

**Issue by KEINOS** [box-project/box#459]
### TL; DR

Add some kind of **CI workflow to test the archived `phar` function-ability on Windows** over virtual env. GitHub Actions may be?

### TS; DR

Since I'm a use-to-be-windows7 user and shifted to Docker-based development on Mac to run my scripts on a container for my Windows users, I have no Windows environment.

Though, I always wanted to be certain if my Phar file works on the Windows environment locally. 

Because having a local Windows PC runnable Phar a  is a good thing and not all users are allowed to use WSL2 even [they can](https://devblogs.microsoft.com/commandline/wsl2-will-be-generally-available-in-windows-10-version-2004/).

As we all know, do to the lack of Windows testers, we can not ensure all the function-ability of BOX3 on Windows.

But if we can **add minimum tests on a virtual Windows env in our CI** like below, it might be a good start to relieve our Windows worry-ness a little.

1. Run `bin/box.bat` on virtual Windows env and create a simple `phar` file.
2. Run a simple unit test for the created `phar` on virtual Win env.
3. If the test passes then exit the CI with `0` status code.

To do so, I think we have 2 options so far.

1. Use the [official Vagrant box image of Win10](https://app.vagrantup.com/Microsoft/boxes/EdgeOnWindows10) and run on TravisCI.
2. Use the [official GitHub-hosted runners of Windows Server 2019](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources) and run on GitHub Actions.

Before any PR, I did some research and it seems that it might be possible.

- Travis CI
  - REFS
    - [Launching a Vagrant VM inside Travis-CI](https://stackoverflow.com/questions/37307114/launching-a-vagrant-vm-inside-travis-ci) @ StackOverflow 
    - [Example project showing how to run Vagrant on TravisCI using libvrt & KVM](https://github.com/jonashackt/vagrant-travisci-libvrt) @ GitHub
    - [Microsoft/EdgeOnWindows10](https://app.vagrantup.com/Microsoft/boxes/EdgeOnWindows10) @ Vagrant Cloud
  - PROS
    - We can stick to the same current main CI such as TravisCI.
    - Little old but Windows 10 and official.
  - CONS
    - The `travis.yml` might get complicated and hard to maintain.
    - No box image update for 4 years. It will take time to `Windows Update` before the unit test.
    - Need to write a script to install PHP.

- GitHub Actions
  - REFS
    - [Write Your GitHub Actions Workflow for Build Windows Application](https://medium.com/rkttu/write-your-github-actions-workflow-for-build-windows-application-94e5a989f477) @ Medium
    - [Github Action for PHP packages](https://dev.to/shivammathur/github-action-for-php-packages-2pii) @ dev.to
    - [Supported runners and hardware resources](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources) | Virtual environments for GitHub-hosted runners @ GitHub Help
  - PROS
    - We can focus on Windows-specific function-ability testing. 
      - Use TravisCI for general testing and use GitHub Actions for [Windows and even for macOS for free](https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions#about-billing-for-github-actions).
    - `latest` tag is available for the Windows image runner.
    - There's a [PHP setup action](https://github.com/marketplace/actions/setup-php-action) on the market place.
    - Microsoft owns GitHub now so hopefully, it will be more Windows-friendly.
  - CONS
    - 2 CIs to handle. Cumbersome procedure it might get.
    - I've no experience with GitHub Actions.
    - Windows Server 2019 and not Win 10.

My opinion is to implement a simple CI with GitHub Actions. But before drafting a PR for this feature **I would like to hear your opinions**.

What do you think?

**Comment by theofidry** [comment]
There is also AppVeyor: as per https://github.com/humbug/box/issues/193 I'm fine with enabling it but since I don't use Windows myself its support will be community based :)

**Comment by KEINOS** [comment]
> since I don't use Windows myself its support will be community based :)

I agree, and it should be so.

It would be nice if it increases Windows' contributors a little more :)

I did a simple test of GitHub Actions.

- [Sample repo to compile and run on Windows env via GitHub Actions](https://github.com/KEINOS/Practice_GitHub_Action-PHP_BOX3) @ GitHub

And it was much easier than I thought. By just adding a single YAML file like below, we can run the tests on Windows.

- [./github/workflows/php-win.yml](https://github.com/KEINOS/Practice_GitHub_Action-PHP_BOX3/blob/2b5f9600c87be7dea45ca22b0fbdd2737a88e4b0/.github/workflows/php-win.yml)

So I would prefer to use GitHub Actions rather than TravisCI for Windows testing.

**Comment by owenvoke** [comment]
@KEINOS, is that because there is a missing quote for the `7.4` ref?
https://github.com/KEINOS/Practice_GitHub_Action-PHP_BOX3/blob/1233d1ab3dec6c134e7dbd92f1cc47d537427d3b/.github/workflows/php-win.yml#L8
```diff
-php-versions: ["7.2", "7.3", 7.4']
+php-versions: ["7.2", "7.3", '7.4']
```

**Comment by KEINOS** [comment]
@owenvoke 

>  is that because there is a missing quote for the 7.4 ref?

O.M.G.!    Σ(￣□￣；）

**Comment by owenvoke** [comment]
Easy mistake to make, and frustrating. 😬

**Comment by theofidry** [comment]
@KEINOS sorry for the late reply. I'll happily accept a new Windows build (although will not set it as required) if one wants to provide one :)
